lib/_http_agent.js

```diff
@@ -481,9 +482,15 @@ Agent.prototype.keepSocketAlive = function keepSocketAlive(socket) {
       const hint = /^timeout=(\d+)/.exec(keepAliveHint)?.[1];
 
       if (hint) {
-        const serverHintTimeout = NumberParseInt(hint) * 1000;
-
-        if (serverHintTimeout < agentTimeout) {
+        // Let the timer expire before the announced timeout to reduce
+        // the likelihood of ECONNRESET errors
+        let serverHintTimeout = (NumberParseInt(hint) * 1000) - 1000;
+        serverHintTimeout = serverHintTimeout > 0 ? serverHintTimeout : 0;
```

Comment:
You can simplify and potentially improve this code by checking if hint bigger than 0 or !NaN and just return 0 as timeout before calling parseInt and multiplying it with 1000.

Reply:
I agree with you that this can be simplified, but I am not sure if the `NumberParseInt` can be early avoided. The `hint` comes back from regex, and is always a string. It will need to be checked with the `1000` before going to the conditions

Reply:
I had a test case failing after the node 20.18.0 upgrade, where the server's keep-alive timeout was set to 1000 ms, but with this change it's reduced by 1 second, so by setting it to <= 1000 ms you now effectively configure it as 0 which then effectively disable keep-alive, if i understood the code correctly?

Reply:
Yes, that's because of this commit. Your understanding is correct.

There is a TODO to expose this as an configurable option. I just need to find time to do that, or anyone who would like to give it a swing is more than welcome to raise a PR. : )